### PR TITLE
Shader configuration improvements

### PR DESCRIPTION
**Pull request by ahigerd**
Killing four birds with one stone:

* Shader uniforms haven't saved since 2017 because the code was written to assume the shader settings had its own OK button. I could have addressed this with a direct function call but I thought a `saveSettingsRequested()` signal would allow future settings panels to be self-contained.
* Unlike basically everything else in the settings window, loading a shader saves the user's choice immediately. This means the Cancel button doesn't undo the action. This PR will still apply the shader changes immediately but it won't save the settings to disk until the user clicks OK or Apply, and the shader changes will be reverted upon clicking Cancel.
* It's kind of annoying to work on shader development with a packaged release of mGBA because you can't select an unpacked shader and you don't have the build system to pack it for you. With this PR you can now select the manifest.ini out of an unpacked shader and it'll load just fine.
* Shader load errors fail silently in the current version of mGBA. This PR adds a simple warning message.

### CODE REVIEW

File: src/platform/qt/ShaderSelector.cpp [hunk 1]
```diff
@@ -66,33 +89,47 @@ void ShaderSelector::selectShader() {
 #if !defined(USE_LIBZIP) && !defined(USE_MINIZIP)
 	QString name = GBAApp::app()->getOpenDirectoryName(this, tr("Load shader"), path.absolutePath());
 #else
-	QString name = GBAApp::app()->getOpenFileName(this, tr("Load shader"), "mGBA Shaders (*.shader)", path.absolutePath());
+	QString name = GBAApp::app()->getOpenFileName(this, tr("Load shader"), "mGBA Shaders (*.shader manifest.ini)", path.absolutePath());
```
Comment (ahigerd) [inline src/platform/qt/ShaderSelector.cpp:92]: Hmm... This really ought to be `QStringLiteral("%1 (*.shader manifest.ini)").arg(tr("mGBA Shaders"))`. Might as well work that in.

File: src/platform/qt/ShaderSelector.h [hunk 2]
```diff
@@ -28,17 +28,19 @@ Q_OBJECT
...
 	void selectShader();
-	void loadShader(const QString& path);
-	void clearShader();
+	void loadShader(const QString& path, bool saveToSettings = false);
+	void clearShader(bool saveToSettings = false);
 	void buttonPressed(QAbstractButton*);
 
 signals:
-	void saved();
+	void saveSettingsRequested();
```
Comment (ahigerd) [inline src/platform/qt/ShaderSelector.h:43]: Strictly speaking, just a cosmetic change, but I think it more accurately reflects the semantics of the signal.

File: src/platform/qt/ShaderSelector.cpp [hunk 3]
```diff
@@ -66,33 +89,48 @@ void ShaderSelector::selectShader() {
 #if !defined(USE_LIBZIP) && !defined(USE_MINIZIP)
 	QString name = GBAApp::app()->getOpenDirectoryName(this, tr("Load shader"), path.absolutePath());
 #else
-	QString name = GBAApp::app()->getOpenFileName(this, tr("Load shader"), "mGBA Shaders (*.shader)", path.absolutePath());
+	QString filters = QStringLiteral("%1 (*.shader manifest.ini)").arg(tr("mGBA Shaders")),
```
Comment (endrift) [inline src/platform/qt/ShaderSelector.cpp:92]:
```suggestion
	QString filters = QStringLiteral("%1 (*.shader manifest.ini)").arg(tr("mGBA Shaders"));
```

File: src/platform/qt/ShaderSelector.h (hunk 2 above)
Comment (endrift) [reply src/platform/qt/ShaderSelector.h:43]: Sure, sounds good.

File: src/platform/qt/ShaderSelector.cpp (hunk 3 above)
Comment (ahigerd) [reply src/platform/qt/ShaderSelector.cpp:92]: Now how did that happen? I compiled and ran this. O.o; Did I fat-finger a vim command while preparing the PR?

**Pull request merged**